openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -1,854 +1,131 @@
-            tenant_to_name[p.get('id')] = name
-
-        cached_servers = self.servers_cache.get('servers')
-        # NOTE: updated_time need to be set at the beginning of this method in order to no miss servers changes.
-        changes_since = datetime.utcnow().isoformat()
-        if cached_servers is None:
-            updated_servers = self.get_active_servers(tenant_to_name)
-        else:
-            previous_changes_since = self.servers_cache.get('changes_since')
-            updated_servers = self.update_servers_cache(cached_servers, tenant_to_name, previous_changes_since)
-
-        # Filter out excluded servers
-        servers = {}
-        for updated_server_id, updated_server in iteritems(updated_servers):
-            if not any([re.match(rule, updated_server_id) for rule in exclude_server_id_rules]):
-                servers[updated_server_id] = updated_server
-
-        # Initialize or update cache for this instance
-        self.servers_cache = {'servers': servers, 'changes_since': changes_since}
-        return servers
-
-    def collect_server_diagnostic_metrics(self, server_details, tags=None, use_shortname=False):
-        def _is_valid_metric(label):
-            return label in NOVA_SERVER_METRICS or any(seg in label for seg in NOVA_SERVER_INTERFACE_SEGMENTS)
-
-        def _is_interface_metric(label):
-            return any(seg in label for seg in NOVA_SERVER_INTERFACE_SEGMENTS)
-
-        tags = tags or []
-        tags = copy.deepcopy(tags)
-        tags.append("nova_managed_server")
-        hypervisor_hostname = server_details.get('hypervisor_hostname')
-        host_tags = self._get_host_aggregate_tag(hypervisor_hostname, use_shortname=use_shortname)
-        host_tags.append('availability_zone:{}'.format(server_details.get('availability_zone', 'NA')))
-        self.external_host_tags[server_details.get('server_name')] = host_tags
-
-        server_id = server_details.get('server_id')
-        server_name = server_details.get('server_name')
-        hypervisor_hostname = server_details.get('hypervisor_hostname')
-        project_name = server_details.get('project_name')
-
-        try:
-            server_stats = self.get_server_diagnostics(server_id)
-        except InstancePowerOffFailure as e:  # 409 response code came back for nova
-            self.log.debug("Server %s is powered off and cannot be monitored: %s", server_id, e)
-            return
-        except requests.exceptions.HTTPError as e:
-            if e.response.status_code == 404:
-                self.log.debug("Server %s is not in an ACTIVE state and cannot be monitored, %s", server_id, e)
-            else:
-                self.warning(
-                    "Received HTTP Error when reaching the Diagnostics endpoint for server %s: %s", server_name, e
+        self.projects_discovery = None
+        if self.config.projects:
+            config_projects_include = normalize_discover_config_include(self.config.projects, ["name"])
```

Comment:
Can we do this part in the config class so we don't do repeated parsing every check run?

Reply:
This code is executed from `check_initializations`, so it is only executed once

Reply:
Oh yes you're right! I don't know why I thought it was in the check method 🤦‍♀️